src/gui/TrackContentObjectView.cpp

```diff
@@ -1106,6 +1127,122 @@ void TrackContentObjectView::toggleMute( QVector<TrackContentObjectView *> tcovs
 	}
 }
 
+bool TrackContentObjectView::canMergeSelection(QVector<TrackContentObjectView*> tcovs)
+{
+	// We can only merge InstrumentTrack's TCOs, so check if we only have those in the selection,
+	// and also if they all belong to the same track
+
+	// We start assuming we can merge and if any condition is broken we set it to false
+	bool canMerge = true;
+
+	// Variable to check if all TCOs belong to the same track
+	TrackView* previousOwnerTrackView = nullptr;
+
+	// Then we check every selected TCO to see if all of them are InstrumentTrack's TCOs.
+	// If any isn't, we set canMerge to false and quit the loop.
+	for (auto tcov: tcovs)
+	{
+		TrackView* ownerTrackView = tcov->getTrackView();
+
+		// Set the previousOwnerTrackView to the first TrackView
+		if (!previousOwnerTrackView)
+		{
+			previousOwnerTrackView = ownerTrackView;
+		}
+
+		// If there are TCOs from different tracks or TCOs from tracks
+		// other than an InstrumentTrack, can't merge them
+		if (ownerTrackView != previousOwnerTrackView
+			|| !dynamic_cast<InstrumentTrackView*>(ownerTrackView))
+		{
+			canMerge = false;
+			break;
+		}
+	}
+
+	return canMerge;
+}
```

Comment:
Simplify the logic by outsourcing it to a set data structure? Probably no noticeable performance impact.
```suggestion
bool TrackContentObjectView::canMergeSelection(QVector<TrackContentObjectView*> tcovs)
{
	bool canMerge = tcovs.size() > 1 &&
		!dynamic_cast<InstrumentTrackView*>(tcovs.at(0)->getTrackView());

	QSet<TrackView*> parentTracks;
	for (auto tcov: tcovs){ parentTracks.insert(tcov); }

	return canMerge && parentTracks.size() == 1;
}
```

Reply:
Done! Used `std::set` instead and removed the `tcovs.size() > 1` since that was checked for on the `contextMenuAction`. Thinking about moving it here though, probably belongs better to this method.

Reply:
Yeah, that's why I included it, feels incorrect for canMerge to return true for a single clip.

Reply:
Agreed, it wasn't a problem before because canMergeSelection was only called with multiple TCOs, but this check suits the method better. Changed on the last commit!